### PR TITLE
Add optional caching for the bearer token

### PR DESCRIPTION
**Pull request by tijmenb**
This allows users to specify a caching adapter to cache the bearer token.

### CODE REVIEW

File: lib/gds-sso/config.rb [hunk 1]
```diff
@@ -18,6 +20,9 @@ module Config
       mattr_accessor :auth_valid_for
       @@auth_valid_for = 20 * 3600
 
+      mattr_accessor :cache
+      @@cache = ActiveSupport::Cache::NullStore.new
```
Comment (boffbowsh) [inline lib/gds-sso/config.rb]: I'm in two minds about whether this should default to `MemoryStore`. At least then there's some caching by default?

Comment (tijmenb) [inline lib/gds-sso/config.rb]: Yeah, I don't know either. 

The idea behind passing in the adapter is that there's at least some evidence in the client app that caching exists, and the behaviour is not so magic.

https://github.com/alphagov/publishing-api/blob/79e24f3c491fd6af6bab8cb3948648a11f4ce0d7/config/initializers/gds-sso.rb#L6

But perhaps there are other ways of doing that.

Comment (boffbowsh) [inline lib/gds-sso/config.rb]: Yeah ok, that makes sense.

**Comment by boffbowsh** [comment]
👍 so far

**Comment by tijmenb** [comment]
@boffbowsh updated PR with docs & test.

**Pull request merged**